cmd/envoy-gateway/root/root.go

```diff
@@ -11,16 +11,38 @@ import (
 	"github.com/envoyproxy/gateway/internal/cmd"
 )
 
-// GetRootCommand returns the root cobra command to be executed
-// by main.
-func GetRootCommand() *cobra.Command {
+// GetRootCommand returns the root cobra command to be executed by main.
+// This command receives an async error handler to let the main process decide how to
+// handle critical errors that may happen in the runners that may prevent Envoy Gateway from
+// functioning properly.
+// The Envoy AI Gateway CLI is an example use case of this function, where it needs to terminate
+// if the infra runner fails to start the Envoy process.
+func GetRootCommand(asyncErrHandler func(error)) *cobra.Command {
+	errChan := make(chan error)
+	if asyncErrHandler == nil {
+		asyncErrHandler = func(error) {}
+	}
+
 	c := &cobra.Command{
 		Use:   "envoy-gateway",
 		Short: "Envoy Gateway",
 		Long:  "Manages Envoy Proxy as a standalone or Kubernetes-based application gateway",
+		PersistentPreRun: func(cmd *cobra.Command, _ []string) {
+			go func() {
+				for {
+					select {
+					case <-cmd.Context().Done():
+						close(errChan)
```

Comment:
Would this cause panic if other go routines send errors to this channel after it's closed?